src/hooks/useArNSRecordsForWallet.tsx

```diff
@@ -0,0 +1,38 @@
+import { AoArNSNameData } from '@ar.io/sdk';
+import { useGlobalState, useWalletState } from '@src/state';
+import { UseQueryResult, useQuery } from '@tanstack/react-query';
+
+export const useArNSRecordsForWallet = (): UseQueryResult<
+  Record<string, AoArNSNameData>
+> => {
+  const [{ walletAddress }] = useWalletState();
+  const [{ arioContract }] = useGlobalState();
+  return useQuery({
+    queryKey: ['arns-records-for-wallet', walletAddress?.toString()],
```

Comment:
arioProcessId, aoNetwork.ARIO, and hyperbeamUrl should be added to the query key here